### ISSUE TITLE
az deployment group what-if is failing after release 2.24.0

### ISSUE THREAD

**Issue by pavel-kurnosov** [Azure/azure-cli#18235]
## Describe the bug

**Command Name**
`az deployment group what-if`

**Errors:**
```
'str' object has no attribute 'value'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az deployment group what-if -f {} -p {} -g {}`

## Environment Summary
```
macOS-11.4-x86_64-i386-64bit
Python 3.8.10
Installer: HOMEBREW

azure-cli 2.24.0

Extensions:
aks-preview 0.5.14
azure-devops 0.18.0

```

**Comment by nodranael** [comment]
Same behaviour here on Windows. It used to work before 2.24.0.

`--debug` shows the following : 

```
cli.azure.cli.core.util: azure.cli.core.util.handle_exception is called with an exception:
cli.azure.cli.core.util: Traceback (most recent call last):
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 691, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 328, in __call__
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/resource/custom.py", line 752, in what_if_deploy_arm_template_at_resource_group
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/resource/custom.py", line 833, in _what_if_deploy_arm_template_core
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/resource/_formatters.py", line 66, in format_what_if_operation_result
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/resource/_formatters.py", line 111, in _format_change_type_legend
AttributeError: 'str' object has no attribute 'value'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/arm.py", line 100, in handle_template_based_exception
AttributeError: 'AttributeError' object has no attribute 'inner_exception'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 231, in invoke
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 657, in execute
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 720, in _run_jobs_serially
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 712, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/arm.py", line 102, in handle_template_based_exception
knack.util.CLIError: 'str' object has no attribute 'value'

cli.azure.cli.core.azclierror: 'str' object has no attribute 'value'
az_command_data_logger: 'str' object has no attribute 'value'
cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x042800B8>]
```

**Comment by zhoxing-ms** [comment]
@shenglol Hi, may I ask is this PR https://github.com/Azure/azure-cli/pull/18244/files to fix this issue? If so, please add this issue link in your PR.

**Comment by dallmair** [comment]
This is fixed in release 2.24.1, which was published a few hours ago:

https://github.com/Azure/azure-cli/pull/18307
